components/dashboard/src/Login.tsx

```diff
@@ -62,8 +62,11 @@ export function Login() {
 
     const updateUser = async () => {
         await getGitpodService().reconnect();
-        const user = await getGitpodService().server.getLoggedInUser();
-        setUser(user);
+        const [ user, teams ] = await Promise.all([
+            getGitpodService().server.getLoggedInUser(),
+            getGitpodService().server.getTeams(),
+        ]);
+        setContext({ user, teams });
```

Comment:
I think we should rather have a Context per object. Both lifecycle and usages of user and teams seems too different.